packages/plugin-nft-generation/src/actions/mintNFTAction.ts

```diff
@@ -0,0 +1,338 @@
+import {
+    Action,
+    composeContext,
+    Content,
+    elizaLogger,
+    generateObjectDeprecated,
+    HandlerCallback,
+    IAgentRuntime,
+    Memory,
+    ModelClass,
+    State,
+} from "@ai16z/eliza";
+import { createNFT } from "../handlers/createNFT.ts";
+import { verifyNFT } from "../handlers/verifyNFT.ts";
+import { sleep } from "../index.ts";
+import WalletSolana from "../provider/wallet/walletSolana.ts";
+import { PublicKey } from "@solana/web3.js";
+
+const mintTemplate = `Respond with a JSON markdown block containing only the extracted values. Use null for any values that cannot be determined.
+
+Example response:
+\`\`\`json
+{
+    "collectionAddress": "D8j4ubQ3MKwmAqiJw83qT7KQNKjhsuoC7zJJdJa5BkvS",
+}
+\`\`\`
+
+{{recentMessages}}
+
+Given the recent messages, extract the following information about the requested mint nft:
+- collection contract address
+
+Respond with a JSON markdown block containing only the extracted values.
+
+Note: Make sure to extract the collection address from the most recent messages whenever possible.`;
+
+export interface MintContent extends Content {
+    collectionAddress: string;
+}
+
+function isMintNFTContent(
+    runtime: IAgentRuntime,
+    content: any
+): content is MintContent {
+    console.log("Content for mint", content);
+    return typeof content.collectionAddress === "string";
+}
+
+const mintNFTAction: Action = {
+    name: "MINT_NFT",
+    similes: [
+        "NFT_MINTING",
+        "NFT_CREATION",
+        "CREATE_NFT",
+        "GENERATE_NFT",
+        "MINT_TOKEN",
+        "CREATE_TOKEN",
+        "MAKE_NFT",
+        "TOKEN_GENERATION",
+    ],
+    description: "Mint NFTs for the collection",
+    validate: async (runtime: IAgentRuntime, _message: Memory) => {
+        const AwsAccessKeyIdOk = !!runtime.getSetting("AWS_ACCESS_KEY_ID");
+        const AwsSecretAccessKeyOk = !!runtime.getSetting(
+            "AWS_SECRET_ACCESS_KEY"
+        );
+        const AwsRegionOk = !!runtime.getSetting("AWS_REGION");
+        const AwsS3BucketOk = !!runtime.getSetting("AWS_S3_BUCKET");
+
+        return (
+            AwsAccessKeyIdOk ||
+            AwsSecretAccessKeyOk ||
+            AwsRegionOk ||
+            AwsS3BucketOk
+        );
+    },
+    handler: async (
+        runtime: IAgentRuntime,
+        message: Memory,
+        state: State,
+        options: { [key: string]: unknown },
+        callback: HandlerCallback
+    ) => {
+        try {
+            elizaLogger.log("Composing state for message:", message);
+            if (!state) {
+                state = (await runtime.composeState(message)) as State;
+            } else {
+                state = await runtime.updateRecentMessageState(state);
+            }
+
+            // Compose transfer context
+            const transferContext = composeContext({
+                state,
+                template: mintTemplate,
+            });
+
+            const content = await generateObjectDeprecated({
```

Comment:
We should be using generateObject and templated type safe schema's se plugin-coinbase for context

Reply:
Thanksfor your guidance.